conda/core/subdir_data.py

```diff
@@ -219,17 +198,26 @@ def __init__(
 
     @property
     def _repo(self) -> RepoInterface:
-        """Changes as we mutate self.repodata_fn."""
-        return self.RepoInterface(
-            self.url_w_credentials,
-            self.repodata_fn,
-            cache_path_json=self.cache_path_json,
-            cache_path_state=self.cache_path_state,
-        )
+        """
+        Changes as we mutate self.repodata_fn.
+        """
+        return self.repo_fetch._repo
 
     @property
     def repo_cache(self) -> RepodataCache:
-        return RepodataCache(self.cache_path_base, self.repodata_fn)
+        return self.repo_fetch.repo_cache
+
+    @property
+    def repo_fetch(self) -> RepodataFetch:
+        """
+        Most modern "exactly what we need" interface.
```

Comment:
I'm must admit, I am little confused by what this doc string is trying to communicate. Are you trying to say that this is the preferred way to instantiate a `RepodataFetch` object?

Reply:
We started with `fetch_repodata_remote_request` (at the end of the file), then we moved through `_repo`, `repo_cache` and finally `repo_fetch`, all of which remain to support compatibility rules.

Reply:
Okay, should probably mention that in the doc string 😁.